manager/controllers/app/m4dapplication_controller.go

```diff
@@ -108,7 +108,9 @@ func (r *M4DApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	if hasError(applicationContext) {
 		log.Info("Reconciled with errors: " + getErrorMessages(applicationContext))
 	}
-	if !applicationContext.Status.Ready {
+
+	// trigger a new reconcile if required (the m4dapplication is not ready and not empty)
+	if !applicationContext.Status.Ready && (len(applicationContext.Spec.Data) > 0) {
```

Comment:
Why the extra (brackets) in `(len(applicationContext.Spec.Data) > 0)`?

Reply:
Removed.